java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/converter/AvroMessageConverter.java

```diff
@@ -0,0 +1,39 @@
+package com.cumulocity.generic.mqtt.client.converter;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.api.SchemaSerializationException;
+
+/**
+ * A converter to serialize message object to byte array and vice versa.
+ */
+public interface AvroMessageConverter<T> {
+
+    /**
+     * Encode an object representing the message content into a byte array.
+     *
+     * @param message the message object
+     * @return a byte array with the serialized content
+     * @throws SchemaSerializationException if the serialization fails
+     */
+    byte[] encode(T message);
+
+    /**
+     * Decode a byte array into an object using the schema definition and deserializer implementation.
+     *
+     * @param bytes the byte array to decode
+     * @return the deserialized object
+     */
+    T decode(byte[] bytes);
+
+    /**
+     * Create an Avro schema type by default configuration of the class.
+     *
+     * @param <T>   This is the type parameter
+     * @param clazz the POJO class to be used to extract the Avro schema
```

Comment:
issue: we should introduce the POJO aspect in the interface description if it must be a POJO (a simple data object) if we introduce it at all.
suggestion:  clazz class used to generate the Avro schema.